distribution/push.go

```diff
@@ -184,3 +186,16 @@ func compress(in io.Reader) (io.ReadCloser, chan struct{}) {
 
 	return pipeReader, compressionDone
 }
+
+// return the correct compressor based on the given number of threads,
+// wrapping the given writer.
+func newCompressor(bufWriter io.Writer, compressionThreads int) io.WriteCloser {
```

Comment:
What should it do if the number is negative, return an error? Currently, it will just silently use the standard gzip library.

If that's what the concern is, I would also validate that the number is positive when the flag if first parsed, so the user can be alerted before things get too far.